tests/integration/targets/aci_l3out_node_profile/tasks/main.yml

```diff
@@ -0,0 +1,173 @@
+    <<: *aci_info
+    state: present
+    tenant: ansible_tenant
+    l3out: ansible_l3out
+    domain: ansible_dom
+    route_control: export
+    vrf: ansible_vrf
+    l3protocol: ospf
+  register: l3out_present
+
+- name: Add L3out node profile (check mode)
+  cisco.aci.aci_l3out_node_profile: &aci_node_profile_present
+    <<: *aci_info
+    node_profile: ansible_node_profile
+    description: 'Node profile for Ansible testing'
+    dscp: CS0
+    l3out: ansible_l3out
+    tenant: ansible_tenant
+    state: present
+  check_mode: yes
+  register: cm_add_node_profile
+
+- name: Add L3out node profile (normal mode)
+  cisco.aci.aci_l3out_node_profile: *aci_node_profile_present
+  register: nm_add_node_profile
+
+- name: Verify add_node_profile
+  assert:
+    that:
+    - cm_add_node_profile is changed
+    - nm_add_node_profile is changed
+    - cm_add_node_profile.sent.l3extLNodeP.attributes.descr == nm_add_node_profile.sent.l3extLNodeP.attributes.descr == 'Node profile for Ansible testing'
+    - cm_add_node_profile.sent.l3extLNodeP.attributes.name == nm_add_node_profile.sent.l3extLNodeP.attributes.name == 'ansible_node_profile'
+    - cm_add_node_profile.proposed.l3extLNodeP.attributes.descr == nm_add_node_profile.proposed.l3extLNodeP.attributes.descr == 'Node profile for Ansible testing'
+    - cm_add_node_profile.proposed.l3extLNodeP.attributes.name == nm_add_node_profile.proposed.l3extLNodeP.attributes.name == 'ansible_node_profile'
+    - cm_add_node_profile.current == cm_add_node_profile.previous == nm_add_node_profile.previous == []
+    - nm_add_node_profile.current.0.l3extLNodeP.attributes.dn == 'uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile'
+    - nm_add_node_profile.current.0.l3extLNodeP.attributes.name == 'ansible_node_profile'
+
+- name: Modify node profile (check mode)
+  cisco.aci.aci_l3out_node_profile: &aci_node_profile_modify
+    <<: *aci_info
+    node_profile: ansible_node_profile
+    description: 'my_updated_descr'
+    l3out: ansible_l3out
+    tenant: ansible_tenant
+    state: present
+  check_mode: yes
+  register: cm_mod_node_profile
+
+- name: Modify node profile (normal mode)
+  cisco.aci.aci_l3out_node_profile: *aci_node_profile_modify
+  register: nm_mod_node_profile
+
+- name: Verify mod_node_profile
+  assert:
+    that:
+    - cm_mod_node_profile is changed
+    - nm_mod_node_profile is changed
+    - cm_mod_node_profile.sent.l3extLNodeP.attributes.descr == nm_mod_node_profile.sent.l3extLNodeP.attributes.descr == 'my_updated_descr'
+    - cm_mod_node_profile.proposed.l3extLNodeP.attributes.descr == nm_mod_node_profile.proposed.l3extLNodeP.attributes.descr == 'my_updated_descr'
+    - nm_mod_node_profile.current.0.l3extLNodeP.attributes.dn == 'uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile'
+    - nm_mod_node_profile.current.0.l3extLNodeP.attributes.name == 'ansible_node_profile'
+
+- name: Query existing node profile (check mode)
```

Comment:
Could it be possible that we create more than one node profiles and query all node profiles there?